api/access/server.go

```diff
@@ -185,3 +242,24 @@ func (s *Server) GetNetworkParameters(ctx context.Context, in *access.GetNetwork
 func (s *Server) GetLatestProtocolStateSnapshot(ctx context.Context, in *access.GetLatestProtocolStateSnapshotRequest) (*access.ProtocolStateSnapshotResponse, error) {
 	return nil, errors.New("not implemented")
 }
+
+func eventsFromBlock(header *flow.Header, ee []flow.Event, blockID flow.Identifier, height uint64) (*access.EventsResponse_Result, error) {
```

Comment:
Overall, I find this function a bit unnecessary, especially since when we have the `blockID`, we already have `blockID` and `height` and only need the header timestamp converted. For the `height` case, we can just get the `header` for both `blockID` and `timestamp`, but why wrap it into such a confusingly named function with so many parameters? It hardly even saves lines of code.

Reply:
I think it's not without merit since the event "translation" code is shared between `GetEventsForBlockIDs()` and `GetEventsForHeightRange()`. But I do think it's enough to take the header and events as the two parameters.

Or just have it as a unexported method of the server that will take a height as a parameter and do the header/events retrieval inside too.

Reply:
I removed the helper, also not fond of it for the limited amount of LoC it saves tbh 🤷